tests/test_rspq.c

```diff
@@ -238,14 +244,16 @@ void test_rspq_flush(TestContext *ctx)
     test_ovl_init();
     DEFER(test_ovl_close());
 
+    // This is meant to verify that the fix in rspq_flush actually
+    // prevents the race condition (see the comment in that function).
+    // If the race condition does happen, this test will fail very quickly.
     uint32_t t0 = TICKS_READ();
-    while (TICKS_DISTANCE(t0, TICKS_READ()) < TICKS_FROM_MS(10000)) {
+    while (TICKS_DISTANCE(t0, TICKS_READ()) < TICKS_FROM_MS(1000)) {
```

Comment:
Is 1 sec enough for it to fail consistently? that's a 10 fold decrease from the older value. Is is the randomness that improves it?

Reply:
Yes, the comment says that if we remove the race condition workaround in rspq_flush, the bug triggers very fast, which is also what I've measured. 1s is even still too much for this test.

You could try and verify it yourself, though.